### PR TITLE
protocols/gossipsub: Revert back to wasm_timer for interval

### PR DESCRIPTION
**Pull request by AgeManning**
As described in https://github.com/libp2p/rust-libp2p/issues/2497 the current implementation for Interval in gossipsub can cause some serious issues with how gossipsub functions. 

The heartbeat interval can grow unbounded, which unbounds the memory cache, exploding RAM and causes messages to be gossiped much later than they should be which can lead to very old messages being re-propagated on networks. 

This PR reverts the Interval implementation back to `wasm_timer`. This should be sufficient to get gossipsub to work as expected however it was noted that this is not as performant as a straight `tokio::time::Interval`. A future PR should be made that ports the interval to be `tokio` agnostic, yet equally performant and compatible with WASM on NodeJS.

### CODE REVIEW

**Review by mxinden — COMMENTED** [review]
Looks good to me. Thanks for reverting. Could you include a changelog entry as well?

Also I think this needs to use `wasm_timer::Instant` instead of `instant::Instant`.


//CC @wngr

**Comment by divagant-martian** [comment]
> Also I think this needs to use `wasm_timer::Instant` instead of `instant::Instant`.

https://github.com/libp2p/rust-libp2p/pull/2245 made the `ProtocolsHandler` trait dependent on `instant`

**Comment by divagant-martian** [comment]
I've removed support for gossipsub in the wasm targets. It's sad since before https://github.com/libp2p/rust-libp2p/pull/2245 there was support for the other two targets, but the `instant::Instant` vs `wasm_timer::Instant` incompatibility makes this the only reasonable solution that I can see for now.

**Comment by divagant-martian** [comment]
Following @mxinden suggestion, Gossipsub should now be removed only in the `wasm32-unknown-unknown` target. I'm using `cargo tree` to verify this:
```bash
> cargo tree -e features --target=wasm32-unknown-unknown | rg gossipsub
# empty
> cargo tree -e features --target=wasm32-wasi | rg gossipsub
#├── libp2p-gossipsub feature "default"
#│   └── libp2p-gossipsub v0.36.0 (/home/me/rust-libp2p/protocols/gossipsub)
#│       ├── libp2p-gossipsub feature "default" (*) # libp2p-metrics dep
> cargo tree -e features --target=wasm32-unknown-emscripten | rg gossipsub
#├── libp2p-gossipsub feature "default"
#│   └── libp2p-gossipsub v0.36.0 (/home/me/rust-libp2p/protocols/gossipsub)
#│       ├── libp2p-gossipsub feature "default" (*) # libp2p-metrics dep
```

File: protocols/gossipsub/CHANGELOG.md [hunk 1]
```diff
@@ -10,9 +10,12 @@
 
 - Merge NetworkBehaviour's inject_\* paired methods (see PR 2445).
 
+- Revert to wasm-timer (see PR 2506).
```
Comment (mxinden) [inline protocols/gossipsub/CHANGELOG.md]:
```suggestion
- Revert to wasm-timer (see [PR 2506]).
```

**Review by mxinden — APPROVED** [review]

**Comment by mxinden** [comment]
Thanks @divagant-martian and @AgeManning!

**Pull request merged**